presto-main/src/main/java/com/facebook/presto/type/VarcharOperators.java

```diff
@@ -150,6 +150,19 @@ public static double castToDouble(@SqlType("varchar(x)") Slice slice)
 
     @LiteralParameters("x")
     @ScalarOperator(CAST)
+    @SqlType(StandardTypes.FLOAT)
+    public static long castToFloat(@SqlType("varchar(x)") Slice slice)
```

Comment:
Correct. AST nodes should describe syntactic structures. Since there's no syntactic difference between a float and a double, it doesn't make sense to have separate AST nodes. Eventually, what we should have nodes for the following types:
* Exact integral number (i.e., only digits)
* Exact decimal number (i.e., digits with a decimal separator)
* Approximate number (i.e., scientific notation)
* Type constructor invocation (anything that looks like `XXXXX 'yyyy'`)